WCFDataService/Client/System/Data/Services/Client/Materialization/HttpWebResponseMessage.cs

```diff
@@ -87,14 +88,15 @@ public HttpWebResponseMessage(HttpWebResponse httpResponse)
         /// <param name="headers">The headers.</param>
         /// <param name="statusCode">The status code.</param>
         /// <param name="getResponseStream">A function returning the response stream.</param>
-        internal HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<Stream> getResponseStream)
+        internal HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<Stream> getResponseStream, IODataResponseMessage underlyingResponseMessage)
```

Comment:
Yes – good idea, I’ll fix this.

Regards
Uffe

From: ZOE [mailto:notifications@github.com]
Sent: 5. september 2016 09:51
To: OData/odata.net <odata.net@noreply.github.com>
Cc: Lauesen.Uffe UFE <UFE@kmd.dk>; Mention <mention@noreply.github.com>
Subject: Re: [OData/odata.net] V3 client avoid stream copy for synchroneous requests. (#656)


In WCFDataService/Client/System/Data/Services/Client/Materialization/HttpWebResponseMessage.cs<https://github.com/OData/odata.net/pull/656#discussion_r77481550>:

> @@ -87,14 +88,15 @@ public HttpWebResponseMessage(HttpWebResponse httpResponse)

>          /// <param name="headers">The headers.</param>

>          /// <param name="statusCode">The status code.</param>

>          /// <param name="getResponseStream">A function returning the response stream.</param>

> -        internal HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<Stream> getResponseStream)

> +        internal HttpWebResponseMessage(HeaderCollection headers, int statusCode, Func<Stream> getResponseStream, IODataResponseMessage underlyingResponseMessage)

getResponseStream and underlyingResponseMessage are kind of dup. Why not add an overloaded contructor to accept underlyingResponseMessage only, so other caller does not need change.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/OData/odata.net/pull/656/files/d4141362bcffd86ed7d5ceb80234c687501b726c#r77481550>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AKq-spbIIs-u5CeIeCFH4_xcoDnlNE8jks5qm8ncgaJpZM4JZoB4>.

Vi gør opmærksom på, at denne e-mail kan indeholde fortrolig information. Hvis du ved en fejltagelse modtager e-mailen, beder vi dig venligst informere afsender om fejlen ved at bruge svarfunktionen. Samtidig beder vi dig slette e-mailen i dit system uden at videresende eller kopiere den. Selvom e-mailen og ethvert vedhæftet bilag efter vores overbevisning er fri for virus og andre fejl, som kan påvirke computeren eller it-systemet, hvori den modtages og læses, åbnes den på modtagerens eget ansvar. Vi påtager os ikke noget ansvar for tab og skade, som er opstået i forbindelse med at modtage og bruge e-mailen.

Please note that this message may contain confidential information. If you have received this message by mistake, please inform the sender of the mistake by sending a reply, then delete the message from your system without making, distributing or retaining any copies of it. Although we believe that the message and any attachments are free from viruses and other errors that might affect the computer or it-system where it is received and read, the recipient opens the message at his or her own risk. We assume no responsibility for any loss or damage arising from the receipt or use of this message.